Automattic/wp-calypso
# Append logmein param to links on click

#### Changes proposed in this Pull Request

* Extension of work in #52909 
* Adds onClick intercept for clicks on links, if the link is pointing at a simple site with a mapped domain it swaps out the url for one including a `?logmein=direct` param.
* Adds replaces urls in the same way for `navigate` calls

#### Testing instructions
![Screenshot(18)](https://user-images.githubusercontent.com/811776/123238446-d6d74000-d521-11eb-8ccd-751caefa9e9b.png)

* Find links in calypso that link out to simple sites with mapped domains.
* "Visit site" on the "My Home" page works for this, so do the domain pages under Plans > Domains
* If the domain is a mapped domain against a simple site,
  * You should see a couple of effects:
    * It should take you through the redirect login flow and land you logged in to the site. (check wordpress_logged_in cookie is set against the domain directly or [check the redirect flow](https://chrome.google.com/webstore/detail/link-redirect-trace/nnpljppamoaalgkieeciijbcccohlpoh?hl=en))
    * After clicking one of these links the links `href` will have been updated (if your link has target=_blank).
* Any other site types (atomic, jetpack) should work as normal.

Unit tests:
```
yarn test-client client/lib/logmein
```

Related to #52741 / #52740 / #52763
